pxr/usd/usdUtils/testenv/testUsdUtilsValidators.cpp

```diff
@@ -114,11 +115,33 @@ TestPackageEncapsulationValidator()
     TF_AXIOM(errors.empty());
 }
 
+static
+void
+TestLayerFileFormatValidator()
+{
+    UsdValidationRegistry& registry = UsdValidationRegistry::GetInstance();
+
+    // Verify the validator exists
+    const UsdValidator *validator = registry.GetOrLoadValidatorByName(
+            UsdUtilsValidatorNameTokens->layerFileFormatValidator);
+
+    TF_AXIOM(validator);
+
+    // Create an empty layer with a core extension
+    const SdfLayerRefPtr passLayer = SdfLayer::CreateNew("pass.usda");
```

Comment:
@dgovil @tallytalwar I wanted your opinions on what a reasonable failing test would be for this validation rule. It seems to me that I would need to install some other file format plugin in order for Sdf to allow me to use some non core usd extension for my layer, however then shipping and installing that plugin just to make this test better seems unreasonable. Any ideas?

Reply:
You could just rename a USDC file with an arbitrary extension, and then use the zipfile writer to author a file with it in it. I don't think your test needs to actually read the file within the zip though, just needs to use the zipfile reader to check the file names

Reply:
That seems to work for the `ARKitFileExtensionChecker` which is helpful for when I get to that one however now I'm a little confused on the effective difference between `ARKitFileExtensionChecker`([complianceChecker code](https://github.com/PixarAnimationStudios/OpenUSD/blob/release/pxr/usd/usdUtils/complianceChecker.py#L925)) and `ArKitLayerChecker` ([complianceChecker code](https://github.com/PixarAnimationStudios/OpenUSD/blob/release/pxr/usd/usdUtils/complianceChecker.py#L766)). 

[Here](https://github.com/beersandrew/usd-test-assets/tree/main/Validation/examples/ARLayerChecker) is my hello world example that triggers `ARKitFileExtensionChecker`, but I'm struggling to do something similar for specifically the `ArKitLayerChecker` since it seems to need to get an `SdfLayer` first.

Reply:
Weird. Those look identical to me. 

I guess taking a step back: 
- The files must have extensions that match the usdz spec
- The first file must be a valid usd type

I think they could be the same test to be honest?

Reply:
I agree and think only `ARKitFileExtensionChecker` is necessary here, will add that PR tomorrow. 

I'll leave this PR open in case @tallytalwar or anyone else has a strong opinion to keep this / modify it in some way. 

Thanks!

Reply:
https://github.com/PixarAnimationStudios/OpenUSD/pull/3368

Reply:
Isn't the `ArKitFileExtensionChecker` also checking for image file extensions? (So seems like `ArKitLayerChecker` is a subset of `ArKitFileExtensionChecker`)? Maybe we can keep just the FileExtensionChecker and it return separate UsdValidationError to cater to both the layer (usda, usdc, usd, usdz) and bundled image extensions ("exr", "jpg", "jpeg", "png")?

Reply:
Right, looks like you doing just that in #3368, just split the ValidationError to cater to layer extensions and image extensions?

I am fine with closing this PR (will leave it to you @beersandrew :D)

Reply:
Yep this makes sense to me. I will:

1. close out this PR
2. adjust https://github.com/PixarAnimationStudios/OpenUSD/pull/3368 to have 2 errors 
3. ignore porting over the [TextureChecker](https://github.com/orgs/PixarAnimationStudios/projects/3/views/1?pane=issue&itemId=65424815)

Thank you both!